### ISSUE TITLE
ava@next failed to recognize babel (v7) settings for React

### ISSUE THREAD

**Issue by jimzhan** [avajs/ava#1879]
## Prerequisites

- `ava` - 1.0.0-beta.6
- `babel` - 7.0.0-beta.54
- ` react`, `react-dom` - v16.4.1

## Description

`ava@next` failed to recognize `babel` (v7) settings.

### Test Source

```js
import test from 'ava'
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import App from '@webapp/ui/src/App'

Enzyme.configure({ adapter: new Adapter() })

test('<App />', (t) => {
  const wrapper = shallow(
    <App>
      Application
    </App>,
  )
  t.pass(wrapper)
})
```

### Error Message & Stack Trace

```
yarn run v1.7.0
$ yarn workspace @webapp/test test
$ npx ava


  1 uncaught exception

  Uncaught exception in ui/App.test.js

  /Users/jim/github.com/webapp/node_modules/pirates/lib/index.js:83

  /Users/jim/github.com/webapp/packages/webapp-ui/src/App.jsx:1
  (function (exports, require, module, __filename, __dirname) { import React from 'react'
                                                                       ^^^^^

  SyntaxError: Unexpected identifier

  Module._compile (/Users/jim/github.com/webapp/node_modules/pirates/lib/index.js:83:24)
  Object.newLoader [as .jsx] (/Users/jim/github.com/webapp/node_modules/pirates/lib/index.js:88:7)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
Command: /usr/local/Cellar/node/10.6.0/bin/node
Arguments: /usr/local/lib/node_modules/yarn/lib/cli.js test
Directory: /Users/jim/github.com/webapp/packages/webapp-test
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspace for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Config

Copy the relevant section from `package.json`:

```json
{
  "ava": {
    "cache": false,
    "files": [
      "!api/**/*.test.js",
      "ui/**/*.test.js"
    ],
    "require": [
      "@babel/core",
      "@babel/register",
      "@babel/polyfill"
    ],
    "babel": {
      "extensions": ["js", "jsx"],
      "testOptions": {
        "babelrc": false,
        "presets": [
          "@babel/preset-env",
          "@babel/preset-react"
        ]
      }
    }
  }
}
```

### Command-Line Arguments

Copy your npm build scripts or the `ava` command used:

```
yarn test
```

## Relevant Links

- [Source Repository](https://github.com/jimzhan/webapp)
- To reproduce:
  * `git clone https://github.com/jimzhan/webapp`
  * `yarn install`
  * `yarn test`

## Environment

Tell us which operating system you are using, as well as which versions of Node.js, npm, and AVA. Run the following to get it quickly:

```
Node.js v10.6.0
npm v6.2.0
darwin 17.7.0
```

**Comment by jimzhan** [comment]
Revert back to 

- `ava` v0.25
- `babel` v6.26

with the following settings rocks again:

```json
  "ava": {
    "files": [
      "!api/**/*.test.js",
      "ui/**/*.test.js"
    ],
    "require": [
      "babel-register",
      "./setup-env.js"
    ],
    "babel": "inherit"
  },
```

`.babelrc`

```json
{
  "presets": ["env", "react"]
}
```

**Comment by novemberborn** [comment]
The problem seems to be that ES modules aren't transpiled. Looking at the config you provided I can't immediately tell why that'd be the case. I'd love to figure it out though, AVA 0.25 won't receive any patches so you're going to have to upgrade at some point.

**Comment by jimzhan** [comment]
@novemberborn tks for the comment, do you have any working example that I can ref. to?

**Comment by novemberborn** [comment]
Looking at this again, I noticed the error is in `src/App.jsx`, not your test file. You're using `@babel/register` but the only Babel configuration you've shared is in `package.json#ava.babel.testOptions`. I think this means `@babel/register` is running without any options, and thus your `src` files are not really transpiled.

Indeed in your old setup you have a `.babelrc` file which is used by both AVA and Babel.

In short, stick to the `.babelrc` file and you should be fine.

(I'm closing this issue for housekeeping purposes, but let's keep the conversation going.)

**Comment by jimzhan** [comment]
@novemberborn I think this is mainly caused by babel + monorepo, after switching to `jest`, and place exact same `.babelrc` in each package did solve the issue. Now I'm having a homemade package [esnext-scripts](https://github.com/jimzhan/esnext-scripts) to help :-)
